install/kubernetes/cilium/templates/validate.yaml

```diff
@@ -9,3 +9,10 @@
     {{ fail "Hubble Relay requires .Values.hubble.enabled=true" }}
   {{- end }}
 {{- end }}
+
+{{/* validate service monitoring CRDs */}}
+{{- if not (.Capabilities.APIVersions.Has "monitoring.coreos.com/v1") }}
+  {{- if and (.Values.prometheus.enabled) (or (.Values.prometheus.serviceMonitor.enabled) (.Values.operator.prometheus.serviceMonitor.enabled)) }}
+      {{ fail "Service Monitor requires monitoring.coreos.com/v1 CRDs. Please refer to https://github.com/prometheus-operator/prometheus-operator/blob/master/example/prometheus-operator-crd/monitoring.coreos.com_servicemonitors.yaml" }}
```

Comment:
Is pointing towards a master branch the best option here or should we just pick the most recent release and then only bump if we need newer features? Just wondering if it's possible that in future prometheus either moves this example path or replaces it with a v2 CRD or something.

Not such a big deal though.